services/src/main/java/org/keycloak/protocol/oidc/OIDCWellKnownProvider.java

```diff
@@ -182,7 +182,7 @@ public Object getConfig() {
                     .map(ClientScopeModel::getName)
                     .collect(Collectors.toList());
             scopeNames.add(0, OAuth2Constants.SCOPE_OPENID);
-            config.setScopesSupported(scopeNames);
+            config.setScopesSupported(scopeNames.stream().distinct().collect(Collectors.toList()));
```

Comment:
It should be simpler to just check if the set contains the scope and if so, don't add the scope.

Reply:
this is addressed in another PR.
https://github.com/keycloak/keycloak/pull/26540.
kindly review that.